src/hooks/useCart.tsx

```diff
@@ -22,29 +22,67 @@ interface CartContextData {
 const CartContext = createContext<CartContextData>({} as CartContextData);
 
 export function CartProvider({ children }: CartProviderProps): JSX.Element {
+
   const [cart, setCart] = useState<Product[]>(() => {
-    // const storagedCart = Buscar dados do localStorage
+    const storagedCart = localStorage.getItem('@RocketShoes:cart');
 
-    // if (storagedCart) {
-    //   return JSON.parse(storagedCart);
-    // }
+    if (storagedCart) {
+      return JSON.parse(storagedCart);
+    }
 
     return [];
   });
 
+  const getStockByProductId = async (productId: number) => {
+    await api.get(`/stock/${productId}`).then(response => {
+       return (response.data.amount);
+    });
+  }
+
   const addProduct = async (productId: number) => {
     try {
-      // TODO
+      const newCart = [...cart];
+      //const stock = (getStockByProductId(productId)); //
+      const stock = await (await api.get(`/stock/${productId}`)).data.amount;
```

Comment:
Essa estrutura com dois await não está legal.
A requisição pra pegar o produto pode falhar e vc não está fazendo nenhum tratamento pra ela.

Sugiro utilizar um useMemo pra armazenar esse valor. Dessa forma sempre que a condição pra pegar esse valor for atendida, uma nova requisição vai ser feita. Daí vc trata os possíveis erros e retorna o amount da mesma forma que vc está fazendo aí.